### PR TITLE
Datadog log integration

### PR DESCRIPTION
**Pull request by johnfischelli**
### Summary

Adds a new destination to the Logger. We can now send logs to DataDog!

image

### Checklist

- [x] Tested changes end to end
- [ ] Updated documentation
- [ ] Requested one or more reviewers

### CODE REVIEW

File: docs/docs/feature-library/datadog-log-integration.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+This feature allows the following configuration settings.
+
+![datadog-log-integration settings](../../static/img/features/datadog-log-integration/settings.png)
+
+| Setting | Description |
+| --------| ------------|
+| Api key | Your Datadog Account [client token](https://docs.datadoghq.com/account_management/api-app-keys/#client-tokens)
+| Log Level | The minimum log level to send to Datadog. See [Logging](/flex-project-template/building/template-utilities/logging) for more details.
+| Intake Region | The [Datadog Site](https://docs.datadoghq.com/getting_started/site/) for your account.
```
Comment (dremin) [inline docs/docs/feature-library/datadog-log-integration.md:18]: Maybe we could list the valid values that we support for this setting.

File: docs/docs/feature-library/datadog-log-integration.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| Flush Timeout | In milliseconds. Because we send logs to datadog over HTTP - we do not want to make an HTTP request for every log written. This feature will buffer log messages for the flush timeout before making a single HTTP request to Datadog with all buffered log messages. If there are no logs within the Flush timeout, there is not HTTP request to Datadog. |
```
Comment (dremin) [inline docs/docs/feature-library/datadog-log-integration.md:19]: Minor typo `there is not HTTP request to Datadog.` -> `there is no HTTP request to Datadog.`

File: plugin-flex-ts-template-v2/.prettierignore [hunk 3]
```diff
@@ -1 +1,2 @@
-README.md
\ No newline at end of file
+README.md
+*.md
```
Comment (dremin) [inline plugin-flex-ts-template-v2/.prettierignore:2]: ?? :)

File: plugin-flex-ts-template-v2/src/feature-library/datadog-log-integration/destination/Datadog.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if (!this.hasUnsentLogs()) {
+      return;
+    }
+
+    try {
+      await fetch(this.api, {
+        method: 'POST',
+        headers: {
+          'content-type': 'application/json',
+          // 'Content-Encoding': 'gzip',
```
Comment (dremin) [inline plugin-flex-ts-template-v2/src/feature-library/datadog-log-integration/destination/Datadog.ts:65]: Is there a reason this is commented? Should compression be a setting or always enabled/disabled?

Comment (johnfischelli) [reply plugin-flex-ts-template-v2/src/feature-library/datadog-log-integration/destination/Datadog.ts:65]: Datadog recommends you gzip the logs before sending them and I was playing around with it. But I had to include a rather large library to get the gzip function. Just forgot to remove the header! I will do it and resolve this conversation.

File: plugin-flex-ts-template-v2/.prettierignore (hunk 3 above)
Comment (johnfischelli) [reply plugin-flex-ts-template-v2/.prettierignore:2]: When I was editing markdown files prettier was wildly inflating the changeset :D

I will remove this line and resolve.

File: docs/docs/feature-library/datadog-log-integration.md (hunk 1 above)
Comment (johnfischelli) [reply docs/docs/feature-library/datadog-log-integration.md]: Done!

**Review by dremin — APPROVED** [review]
LGTM!

**Pull request merged**